controller/kongplugininstallation/controller.go

```diff
@@ -179,15 +210,41 @@ func (r *Reconciler) listKongPluginInstallationsForSecret(ctx context.Context, o
 		if kpi.Spec.ImagePullSecretRef == nil {
 			continue
 		}
-		if kpi.Spec.ImagePullSecretRef.Namespace == "" {
-			kpi.Spec.ImagePullSecretRef.Namespace = kpi.Namespace
+		ref.EnsureNamespaceInSecretRef(kpi.Spec.ImagePullSecretRef, gatewayv1.Namespace(kpi.Namespace))
+		if err := ref.DoesFieldReferenceCoreV1Secret(*kpi.Spec.ImagePullSecretRef, "imagePullSecretRef"); err != nil {
+			continue
 		}
-		if kpi.Spec.ImagePullSecretRef.Namespace == namespace && kpi.Spec.ImagePullSecretRef.Name == name {
+		if string(*kpi.Spec.ImagePullSecretRef.Namespace) == namespace && string(kpi.Spec.ImagePullSecretRef.Name) == name {
 			recs = append(recs, reconcile.Request{
-				NamespacedName: client.ObjectKey{
-					Name:      kpi.Name,
-					Namespace: kpi.Namespace,
-				},
+				NamespacedName: client.ObjectKeyFromObject(&kpi),
+			})
+		}
+	}
+	return recs
+}
+
+func (r *Reconciler) listReferenceGrantsForKongPluginInstallation(ctx context.Context, obj client.Object) []reconcile.Request {
+	logger := ctrllog.FromContext(ctx)
+
+	grant, ok := obj.(*gatewayv1beta1.ReferenceGrant)
+	if !ok {
+		logger.Error(
+			fmt.Errorf("unexpected object type"),
+			"ReferenceGrant watch predicate received unexpected object type",
+			"expected", "*gatewayapi.ReferenceGrant", "found", reflect.TypeOf(obj),
+		)
+		return nil
+	}
+	var kpiList v1alpha1.KongPluginInstallationList
+	if err := r.Client.List(ctx, &kpiList); err != nil {
+		logger.Error(err, "Failed to list gateways in watch", "referencegrant", grant.Name)
```

Comment:
should the log be `Failed to list KongPluginInstallations in watch` ?

Reply:
Nice catch, I overlooked during copy-paste